src/bsp/platform/esp32/xi_bsp_time_esp32.c

```diff
@@ -56,21 +55,23 @@ void xi_bsp_time_init()
         XI_ESP32_GET_TIME_FROM_RTC( ( time_t* )&sntp_retrieved_time );
     }
 
-    xi_bsp_debug_format( "RTC updated to current datetime: %d", sntp_retrieved_time );
+    xi_bsp_debug_format( "RTC updated to current datetime: %ld", sntp_retrieved_time );
 exit:
     sntp_stop();
 }
 
 xi_time_t xi_bsp_time_getcurrenttime_seconds()
 {
-    time_t current_time = 0;
-    XI_ESP32_GET_TIME_FROM_RTC( &current_time );
+    xi_time_t current_time = 0;
+    XI_ESP32_GET_TIME_FROM_RTC( ( time_t* )&current_time );
```

Comment:
I guess you don't need the time_t* cast anymore here.

Reply:
Good point :+1: